### PR TITLE
Fix exception stacktrace when ulimit is hit OSX.

### PR DESCRIPTION
**Pull request by mikem8361**
Fixes issue #5782.

Add try/catch for FileLoadException and BadImageFormatException around loading/creating StackTraceSymbols class. These are the exceptions that are still throw even when throwOnError: false.

### CODE REVIEW

**Comment by mikem8361** [comment]
@noahfalk and @jkotas review?

**Comment by jkotas** [comment]
Instead of catching this selection of exception types, I think it would be better to catch everything.

**Comment by noahfalk** [comment]
Heh, catching all was the specific pattern I suggested to avoid. However if you think its the right call in this case I'm ok with it.

**Comment by mikem8361** [comment]
I thought you (or maybe it was Noah) didn’t like blindly catching all exceptions.

**Comment by jkotas** [comment]
Right, I do not like it either - but it is sometimes the right thing to do - like in this case.

**Comment by jkotas** [comment]
LGTM

**Comment by mikem8361** [comment]
@jkotas @noahfalk one more time!

**Comment by jkotas** [comment]
I think the reentrancy check is still required. You can get stackoverflow without it.

File: src/mscorlib/src/System/Diagnostics/Stacktrace.cs [hunk 1]
```diff
@@ -64,9 +64,6 @@ internal class StackFrameHelper : IDisposable
         private static Type s_symbolsType = null;
         private static MethodInfo s_symbolsMethodInfo = null;
 
-        [ThreadStatic]
-        private static int t_reentrancy = 0;
```
Comment (noahfalk) [inline src/mscorlib/src/System/Diagnostics/Stacktrace.cs]: It seems safer to leave the reentrancy check in place. Even if we do catch the exceptions, that won't prevent infinite recursion if the code within this block (and its complete call-graph) somewhere throws an exception and then tries to log the stack trace for it.

**Pull request merged**